### PR TITLE
Dockerfile: Bump rustc to get back rustfmt

### PR DESCRIPTION
**Pull request by NickeZ**

### CODE REVIEW

**Comment by x1ddos** [comment]
Would be nice to apply this patch also to `staging-rust-async` branch.

**Comment by benma** [comment]
Thanks. I'll push a new image to dockerhub.

File: Dockerfile [hunk 1]
```diff
@@ -136,7 +136,7 @@ COPY rust-toolchain /tmp/rust-toolchain
 RUN curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | CARGO_HOME=/opt/cargo sh -s -- --default-toolchain $(cat /tmp/rust-toolchain | tr -d '\r\n\t') -y
 RUN rustup target add thumbv7em-none-eabi
 # re-enable and put into CI once available: https://rust-lang.github.io/rustup-components-history/index.html
```
Comment (benma) [inline Dockerfile:138]: delete this line as well?

**Comment by benma** [comment]
It's pushed now, please update the ci image version to `:15`.

**Comment by x1ddos** [comment]
> It's pushed now, please update the ci image version to `:15`.

hm... I've just pulled `:15` but see this error for `rust-bindgen` make target:

```
dockeruser@ab4c2bf26036:bitbox02-firmware/build$ make rust-bindgen
Scanning dependencies of target generate-protobufs
Generating hww.pb.c, backup.pb.c, common.pb.c, backup_commands.pb.c, bitbox02_system.pb.c, bitboxbase.pb.c, btc.pb.c, eth.pb.c, mnemonic.pb.c, random_number.pb.c, system.pb.c, perform_attestation.pb.c, keystore.pb.c, hww.pb.h, backup.pb.h, common.pb.h, backup_commands.pb.h, bitbox02_system.pb.h, bitboxbase.pb.h, btc.pb.h, eth.pb.h, mnemonic.pb.h, random_number.pb.h, system.pb.h, perform_attestation.pb.h, keystore.pb.h
Built target generate-protobufs
Scanning dependencies of target rust-bindgen
error: 'rustfmt' is not installed for the toolchain 'nightly-2020-03-25-x86_64-unknown-linux-gnu'
To install, run `rustup component add rustfmt --toolchain nightly-2020-03-25-x86_64-unknown-linux-gnu`
Failed to run rustfmt: Internal rustfmt error (non-fatal, continuing)
Built target rust-bindgen
```

**Comment by x1ddos** [comment]
Doesn't look like version `:15` picked up rust-toolchain update?

```
$ rustup component add rustfmt
error: component 'rustfmt' for target 'x86_64-unknown-linux-gnu' is unavailable for download for channel nightly-2020-03-25
Sometimes not all components are available in any given nightly.
```

**Comment by x1ddos** [comment]
```
$ docker images | grep firmware
docker.io/shiftcrypto/firmware_v2   15       cc591df65ac6   2 hours ago    4.38 GB
```

**Comment by benma** [comment]
@x1ddos you have to check out this PR too because the toolchain file has to match.

**Comment by x1ddos** [comment]
> @x1ddos you have to check out this PR too because the toolchain file has to match.

oh, of course! yeah, this worked:

```
curl -sSL https://patch-diff.githubusercontent.com/raw/digitalbitbox/bitbox02-firmware/pull/449.patch | patch -p1
dockeruser@ab4c2bf26036:bitbox02-firmware$ rustup show active-toolchain
nightly-2020-04-06-x86_64-unknown-linux-gnu (overridden by '/home/alex/src/bb/bitbox02-firmware/rust-toolchain')
```

I forgot about [override precedence](https://github.com/rust-lang/rustup#override-precedence). This is a bit confusing though. Would it make sense to set `RUSTUP_TOOLCHAIN` when building the container image or at runtime, i.e. when executing `make dockerxxx`?

**Comment by benma** [comment]
> > @x1ddos you have to check out this PR too because the toolchain file has to match.
> 
> oh, of course! yeah, this worked:
> 
> ```
> curl -sSL https://patch-diff.githubusercontent.com/raw/digitalbitbox/bitbox02-firmware/pull/449.patch | patch -p1
> dockeruser@ab4c2bf26036:bitbox02-firmware$ rustup show active-toolchain
> nightly-2020-04-06-x86_64-unknown-linux-gnu (overridden by '/home/alex/src/bb/bitbox02-firmware/rust-toolchain')
> ```
> 
> I forgot about [override precedence](https://github.com/rust-lang/rustup#override-precedence). This is a bit confusing though. Would it make sense to set `RUSTUP_TOOLCHAIN` when building the container image or at runtime, i.e. when executing `make dockerxxx`?

I don't think so; the toolchain should be fixed to the commit, not to the environment you run in. Maybe an exception for development can be done, but I don't think it's worth it for such a rarely changing thing.

**Comment by x1ddos** [comment]
Well, `make dockerdev` runs `scripts/dockerenv.sh` which is also fixed to the commit...

**Comment by NickeZ** [comment]
I think you've run into the issue that `make dockerdev` uses `:latest` and not a tagged docker version.

try run something like:
```
docker tag shiftcrypto/firmware_v2:15 shiftcrypto/firmware_v2:latest
```

**Comment by benma** [comment]
@NickeZ is correct. I usually don't push `:latest`, so tagging it locally will make dockerenv.sh work. I can also push :latest if you want, or we could add the `:15` also to dockerenv.

**Comment by x1ddos** [comment]
I didn't actually use `:latest`. I changed https://github.com/digitalbitbox/bitbox02-firmware/blob/93ae94b72320d2a00c12541abfeef9f524ae7e01/scripts/dockerenv.sh#L57 to `${CONTAINER_IMAGE}:15` before I tried it in https://github.com/digitalbitbox/bitbox02-firmware/pull/449#issuecomment-610302671.

**Comment by benma** [comment]
Merged in https://github.com/digitalbitbox/bitbox02-firmware/pull/452 - thanks!

**Pull request closed without merging**